modules/gui/src/com/haulmont/cuba/gui/components/calendar/ListCalendarEventProvider.java

```diff
@@ -59,6 +60,18 @@ public void removeEvent(CalendarEvent calendarEvent) {
         fireEventSetChange();
     }
 
+    @Override
+    public void removeAllEvents() {
+        Iterator<CalendarEvent> eventIterator = eventList.iterator();
+        while (eventIterator.hasNext()) {
```

Comment:
Simply clear the eventList after iteration. No need to remove each item with iterator:
1. Remove change listener
2. Clear list
3. Fire event